src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -396,6 +397,50 @@ private void VerifyIsAlive()
                 throw new InvalidOperationException(SR.AssemblyLoadContext_Verify_NotUnloading);
             }
         }
+
+        private static readonly AsyncLocal<AssemblyLoadContext> _asyncLocalCurrentContextualReflectionContext = new AsyncLocal<AssemblyLoadContext>();
```

Comment:
This will be rarely used. It would be nice to create it lazily only once something sets it.

(AssemblyLoadContext static constructor is run in pretty much all apps.)

Reply:
I changed the implementation to 
https://github.com/dotnet/coreclr/blob/dcdd98d9f448753f2623ae5109008fc1aaf797fd/src/System.Private.CoreLib/shared/System/Runtime/Loader/AssemblyLoadContext.cs#L401-L410

I believe this will defer the initialization until the `StaticAsyncLocalCurrentContextualReflectionContext` constructor is called, when it is accessed.

Reply:
I was thinking to initialize it only if somebody actually sets the value (do not initialize for get). Like what we do e.g. for current culture: https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs#L447

Reply:
OK seems simple enough and looks obviously thread safe.

However it leads me down a path to figure out what the `!.` operator is...  Seems completely undocumented.
https://github.com/dotnet/coreclr/blob/76a60f410e9703c3350c8fc26c6f5ba077a01743/src/System.Private.CoreLib/shared/System/Globalization/CultureInfo.cs#L451

Reply:
These are the new nullable annotations: https://devblogs.microsoft.com/dotnet/nullable-reference-types-in-csharp/

There are a lot of PRs going in these days to add them everywhere in CoreLib.